src/test/unit/cli_unittest.cc

```diff
@@ -241,38 +241,62 @@ class CliWriteTest : public ::testing::Test
         outLines.clear();
     }
 
+    static void putOutLine(string line) {
+        if (line.empty()) {
+            return;
+        }
+        if (outLines.empty()) {
+            outLines.push_back(line);
+            return;
+        }
+
+        auto &lastLine = outLines.back();
+        if (!lastLine.empty() && lastLine.back() == '\n') {
+            outLines.push_back(line);
+        } else {
+            lastLine.append(line);
+        }
+    }
+
     static void dummyBufWriter(void *, void *_data, int size) {
         uint8_t *data = (uint8_t *)_data;
         ostringstream stream;
         for (int i = 0; i < size; i++) {
             stream << (char)data[i];
         }
-        outLines.push_back(stream.str());
+        auto str = stream.str();
```

Comment:
(a bit shorter version)
```
std::string str(static_cast<const char*>(_data), size);
std::istringstream stream(str);

for (std::string line; std::getline(stream, line)) {
    if (!line.empty() && line.back() == '\r') {
        line.pop_back(); // Remove the '\r' character
    }
    if (!stream.eof()) { // line was terminated by '\n'
        line.push_back('\n');
    }
    putOutLine(line);
}

Reply:
Oh, also started with a stream but did not find out how to detect the last line